crates/rolldown_common/src/types/import_record.rs

```diff
@@ -123,21 +123,55 @@ impl RawImportRecord {
   }
 
   pub fn into_resolved(self, resolved_module: ModuleIdx) -> ResolvedImportRecord {
-    ResolvedImportRecord {
+    let rec = ImportRecord {
       state: ImportRecordStateResolved { resolved_module },
       module_request: self.module_request,
       kind: self.kind,
       namespace_ref: self.namespace_ref,
       meta: self.meta,
       related_stmt_info_idx: self.related_stmt_info_idx,
+    };
+    if resolved_module.is_dummy() {
```

Comment:
Could you add some comments to explain what's dummy modue and when it shows up?

Reply:
updated here https://github.com/rolldown/rolldown/pull/4717